homeassistant/components/blebox/climate.py

```diff
@@ -0,0 +1,87 @@
+"""BleBox climate entity."""
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import BleBoxEntity, create_blebox_entities
+from .const import BLEBOX_TO_CLIMATE_HVAC_MODE, BLEBOX_TO_CURRENT_HVAC_MODE
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up a BleBox climate entity."""
+
+    create_blebox_entities(
+        hass, config_entry, async_add_entities, BleBoxClimateEntity, "climates"
+    )
+
+
+class BleBoxClimateEntity(BleBoxEntity, ClimateDevice):
+    """Representation of a BleBox climate feature (saunaBox)."""
+
+    @property
+    def supported_features(self):
+        """Return the supported climate features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self):
+        """Return the desired HVAC mode."""
+        return BLEBOX_TO_CLIMATE_HVAC_MODE[self._feature.is_on]
```

Comment:
I think we should remove `None` from the lookup source and use the following:

```suggestion
        return BLEBOX_TO_CLIMATE_HVAC_MODE.get(self._feature.is_on)
```